### PR TITLE
F# performance improvements ~75%

### PR DESCRIPTION
**Pull request by dmannock**
## Description
Performance improvements to my previous F# solutions 2 & 3. Gives ~75% boost from changes involving sieve init, inlining & spans.

Test results updated in the readme files.

## Contributing requirements

* [X] I read the contribution guidelines in CONTRIBUTING.md.
* [X] I placed my solution in the correct solution folder.
* [X] I added a README.md with the right badge(s).
* [X] I added a Dockerfile that builds and runs my solution.
* [X] I selected `drag-race` as the target branch.
* [X] All code herein is licensed compatible with BSD-3.

### CODE REVIEW

**Comment by rbergen** [comment]
Now that I'm reviewing this, I'm wondering if it is correct to mark the bit count per prime of these solution as 1. The type that is used to store the prime number flags is bool, which I think means the number of bits per prime is unknown.

**Comment by dmannock** [comment]
Good spot. It appears bits isn't in the output when using BitArray / bool array elsewhere. Omitting it for consistency sounds like a plan 👍

**Comment by rbergen** [comment]
> Good spot. It appears bits isn't in the output when using BitArray / bool array elsewhere. Omitting it for consistency sounds like a plan 👍

The use of bits=1 in solution_1 is accurate, because that implementation uses a BitArray class instead of a bool array. A BitArray guarantees that each item in the class does use one bit. That is to say, there is a relevant difference between the implementations of solution_1 on the one hand, and the other 2 solutions on the other. 

With that in mind, I think that: 
- the bits tag should indeed be removed from the output of solution_2 and solution_3, and the Bits badges in the README.md files of those solutions should be changed to `unknown-yellowgreen`, accordingly. 
- solution_1 should not be modified on this point.

**Comment by dmannock** [comment]
Sorry, my bad. I Intended to update to be consistent with C# solution where thought it defaults to null when using BitArray and so doesn't output bits=1. However, on second glance the implementation that's used in the run all does indeed have BitsPerPrime = 1.  The consistency changes there would better suit another PR if there isn't one already open. I'll sort this PR out with what applies in this context.

File: PrimeFSharp/solution_3/README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 ![Algorithm](https://img.shields.io/badge/Algorithm-base-green)
 ![Faithfulness](https://img.shields.io/badge/Faithful-yes-green)
 ![Parallelism](https://img.shields.io/badge/Parallel-no-green)
-![Bit count](https://img.shields.io/badge/Bits-1-green)
+![Bit count](https://img.shields.io/badge/unknown-yellowgreen)
```
Comment (rbergen) [inline PrimeFSharp/solution_3/README.md]: The full markdown for the Bits badge should be this:

```
![Bit count](https://img.shields.io/badge/Bits-unknown-yellowgreen)
```

Yielding this badge:
Bit count

File: PrimeFSharp/solution_2/README.md [hunk 2]
```diff
@@ -3,7 +3,7 @@
 ![Algorithm](https://img.shields.io/badge/Algorithm-base-green)
 ![Faithfulness](https://img.shields.io/badge/Faithful-yes-green)
 ![Parallelism](https://img.shields.io/badge/Parallel-no-green)
-![Bit count](https://img.shields.io/badge/Bits-1-green)
+![Bit count](https://img.shields.io/badge/unknown-yellowgreen)
```
Comment (rbergen) [inline PrimeFSharp/solution_2/README.md]: See my previous comment about the badge markdown.

File: PrimeFSharp/solution_3/README.md (hunk 1 above)
Comment (dmannock) [reply PrimeFSharp/solution_3/README.md]: Cheers, glad someone has eyes today! 👀

**Comment by rbergen** [comment]
I'm going to merge this as it concerns a scoped change by the original solutions' author.

**Pull request merged**